manifests/jail.pp

```diff
@@ -1,64 +1,101 @@
-  $jail_name               = $name,
-  $user                    = undef,
-  $group                   = undef,
-  $match_group             = undef,
-  $password_authentication = 'no',
+  Enum['present', 'absent'] $ensure                  = 'present',
+  Sftp_jail::File_name      $jail_name               = $name,
+  Stdlib::Absolutepath      $jail_base               = "${sftp_jail::chroot_base}/${jail_name}",
+  Accounts::User::Name      $user                    = $name,
+  Accounts::User::Name      $group                   = $user,
+  Stdlib::Absolutepath      $real_home               = "${jail_base}/home/${user}",
+  Sftp_jail::Sub_dirs       $sub_dirs                = $sftp_jail::sub_dirs,
+  Boolean                   $merge_subdirs           = false,
+  Accounts::User::Name      $match_group             = $group,
+  Boolean                   $password_authentication = false,
 ) {
   include sftp_jail
-  $jail_base = "${sftp_jail::chroot_base}/${jail_name}"
 
-  if ($match_group) {
-    $ssh_match_group = $match_group
-  }
-  else {
-    $ssh_match_group = $group
-  }
-
-  if !($group) {
-    $group = $user
-  }
-
-  file { $jail_base:
-    ensure  => 'directory',
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    require => File[$sftp_jail::chroot_base],
-  }
-
-  file { "${jail_base}/incoming":
-    ensure  => 'directory',
-    owner   => $user,
-    group   => $group,
-    mode    => '0775',
-    require => File[$sftp_jail::chroot_base],
-  }
-
-  file { "${jail_base}/home":
-    ensure  => 'directory',
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    require => File[$sftp_jail::chroot_base],
+  unless $ensure == 'absent' {
+    if $ensure == 'absent' {
+      $_ensure = 'absent'
+    } else {
+      $_ensure = 'directory'
+    }
+    file { [$jail_base, "${jail_base}/home"]:
+        ensure => $_ensure,
+        owner  => 'root',
+        group  => 'root',
+        mode   => '0755',
+    }
+    file { "${jail_base}/incoming":
+      ensure => $_ensure,
+      owner  => $user,
+      group  => $group,
+      mode   => '0775',
+    }
   }
 
-  file { "${jail_base}/home/${user}":
-    ensure  => 'directory',
-    owner   => $user,
-    group   => $group,
-    mode    => '0755',
-    require => File["${jail_base}/home"],
+  sftp_jail::user { $user:
+    ensure        => $ensure,
+    jail          => $jail_base,
+    group         => $group,
+    real_home     => $real_home,
+    sub_dirs      => $sub_dirs,
+    merge_subdirs => $merge_subdirs,
   }
 
-  ssh::server::match_block { $ssh_match_group:
-    type    => 'Group',
-    options => {
-      'ChrootDirectory'        => $jail_base,
-      'ForceCommand'           => 'internal-sftp',
-      'PasswordAuthentication' => $password_authentication,
-      'AllowTcpForwarding'     => 'no',
-      'X11Forwarding'          => 'no',
-    },
+  if $ensure == 'present' {
+    ssh::server::match_block { $match_group:
+      type    => 'Group',
```

Comment:
Is it not possible to ensure this 'absent' as well?

Reply:
also cleaned in https://github.com/voxpupuli/puppet-sftp_jail/pull/71/commits/b4d6fd3d27df6aed2fb5dc7c47edfe7d52430757